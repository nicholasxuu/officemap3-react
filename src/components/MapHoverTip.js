import React, {PropTypes} from 'react';
import * as ImmutablePropTypes from 'react-immutable-proptypes';

class MapHoverTip extends React.Component {

	render = () => {
		return (
			<div>
				MapHoverTip
			</div>
		);
	}
}

// MapHoverTip.propTypes = {
// 	actions: PropTypes.shape({}).isRequired,
// 	posX: PropTypes.number.isRequired,
// 	posY: PropTypes.number.isRequired,
// 	maxWidth: PropTypes.number.isRequired,
// 	maxHeight: PropTypes.number.isRequired,
// };

export default MapHoverTip;

