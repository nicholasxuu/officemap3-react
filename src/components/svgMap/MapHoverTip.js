import React from 'react';
import { PropTypes } from 'react';
import * as ImmutablePropTypes from 'react-immutable-proptypes';
import Immutable from 'immutable';
import '../../styles/svgMap/mapHoverTip.css';
import { Tooltip } from 'react-bootstrap';

class MapHoverTip extends React.Component {

	render = () => {
		if (this.props.show === false) {
			return null;
		}
		return (
			<Tooltip
				id="map-hover-tooltip"
				className="in map-hover-tip"
				placement="top"
			    style={{
			    	left: this.props.clientPosX,
				    top: this.props.clientPosY,
			    }}
			>
					{this.props.location.get('name')}
			</Tooltip>
		);
	}
}

MapHoverTip.defaultProps = {
	show: false,
	clientPosX: 0,
	clientPosY: 0,
	location: Immutable.Map({
		id: 0,
		name: 'dummy',
	}),
};

MapHoverTip.propTypes = {
	show: PropTypes.bool.isRequired,
	clientPosX: PropTypes.number.isRequired,
	clientPosY: PropTypes.number.isRequired,
	location: ImmutablePropTypes.contains({
		id: PropTypes.number,
		name: PropTypes.string,
	}).isRequired,
};

export default MapHoverTip;

