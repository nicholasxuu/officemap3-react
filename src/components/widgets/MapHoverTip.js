import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from 'react-bootstrap';
import * as ImmutablePropTypes from 'react-immutable-proptypes';
import Immutable from 'immutable';
import styled from 'styled-components';
import '../../styles/widgets/mapHoverTip.css';

class MapHoverTip extends React.Component {
  render = () => {
    if (this.props.show === false) {
      return null;
    }

    // if id or name empty, don't display
    if (!this.props.locationObj.has('id') ||
      this.props.locationObj.get('id') === 0 ||
      !this.props.locationObj.has('name') ||
      this.props.locationObj.get('name') === ''
    ) {
      return null;
    }

    return (
      <StyledTooltip
        id="map-hover-tooltip"
        className="in map-hover-tip"
        placement="top"
        left={this.props.clientPosX}
        top={this.props.clientPosY}
      >
        {this.props.locationObj.get('name')}
      </StyledTooltip>
    );
  }
}

const StyledTooltip = styled(Tooltip)`
  left: ${props => props.left}px;
  top: ${props => props.top}px;
  position: fixed;
  zIndex: 2;
  textAlign: center;
  /* prevent causing mouseLeave when mouse touch it */
  pointerEvents: none;
  /* make widget top and center */
  transform: translateX(-50%) translateY(-100%);
`;

MapHoverTip.defaultProps = {
  show: false,
  clientPosX: 0,
  clientPosY: 0,
  locationObj: Immutable.Map({
    id: 0,
    name: '',
  }),
};

MapHoverTip.propTypes = {
  show: PropTypes.bool,
  clientPosX: PropTypes.number,
  clientPosY: PropTypes.number,
  locationObj: ImmutablePropTypes.contains({
    id: PropTypes.number,
    name: PropTypes.string,
  }),
};

export default MapHoverTip;

