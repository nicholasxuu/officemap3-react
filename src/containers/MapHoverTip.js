import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {} from '../actions/map';
import MapHoverTip from '../components/MapHoverTip';

const mapStateToProps = (state) => {
	return {
		show: !state.hoverLocation.get('location').isEmpty(),
		location: state.hoverLocation.get('location'),
		posX: state.hoverLocation.get('posX'),
		posY: state.hoverLocation.get('posY'),
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({}, dispatch),
	}
};

const MapHoverTipContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(MapHoverTip);

export default MapHoverTipContainer;