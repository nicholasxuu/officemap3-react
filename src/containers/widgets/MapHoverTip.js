import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {} from '../../actions/map';
import MapHoverTip from '../../components/widgets/MapHoverTip';

const mapStateToProps = (state) => {
	let show = false;
	if (!state.hoverData.get('location').isEmpty()) {
		show = true;
	}

	return {
		show: show,
		clientPosX: state.hoverData.get('clientPosX'),
		clientPosY: state.hoverData.get('clientPosY'),
		location: state.hoverData.get('location'),
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({}, dispatch),
	}
};

const MapHoverTipReducer = connect(
	mapStateToProps,
	mapDispatchToProps
)(MapHoverTip);

export default MapHoverTipReducer;