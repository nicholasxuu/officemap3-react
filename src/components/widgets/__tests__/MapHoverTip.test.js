/* eslint-disable function-paren-newline,prefer-const */
// Link.react-test.js
import React from 'react';
import Immutable from 'immutable';
import renderer from 'react-test-renderer';
import MapHoverTip from '../MapHoverTip';

test('HoverTip shown', () => {
  const component = renderer.create(
    <MapHoverTip
      show
      clientPosX={10}
      clientPosY={20}
      locationObj={Immutable.fromJS({
        id: 1,
        name: 'locationName',
      })}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('HoverTip not shown', () => {
  const component = renderer.create(
    <MapHoverTip
      show={false}
      clientPosX={10}
      clientPosY={20}
      locationObj={Immutable.fromJS({
        id: 1,
        name: 'locationName',
      })}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toBe(null);
});


test('Invalid HoverTip not shown', () => {
  const component = renderer.create(
    <MapHoverTip
      show
      clientPosX={10}
      clientPosY={20}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toBe(null);
});
