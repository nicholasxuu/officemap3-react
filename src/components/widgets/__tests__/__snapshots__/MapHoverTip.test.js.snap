// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverTip shown 1`] = `
<div
  className="in map-hover-tip tooltip top"
  id="map-hover-tooltip"
  role="tooltip"
  style={
    Object {
      "left": 10,
      "pointerEvents": "none",
      "position": "fixed",
      "textAlign": "center",
      "top": 20,
      "transform": "translateX(-50%) translateY(-100%)",
      "zIndex": 2,
    }
  }
>
  <div
    className="tooltip-arrow"
    style={
      Object {
        "left": undefined,
        "top": undefined,
      }
    }
  />
  <div
    className="tooltip-inner"
  >
    locationName
  </div>
</div>
`;
